pkg/data/resource_metrics.go

```diff
@@ -0,0 +1,356 @@
+package data
+
+import (
+	"fmt"
+	"github.com/golang/glog"
+)
+
+type MetricPropType string
+const (
+	USED    MetricPropType = "Used"
+	CAP     MetricPropType = "Capacity"
+	PEAK    MetricPropType = "Peak"
+	AVERAGE MetricPropType = "Average"
+)
+
+type ResourceType string
+const (
+	CPU      ResourceType = "CPU"
+	MEM      ResourceType = "MEM"
+	DISK     ResourceType = "DISK"
+	MEM_PROV ResourceType = "MEM_PROV"
+	CPU_PROV ResourceType = "CPU_PROV"
+)
+
+
+type EntityType string
+const (
+	NODE EntityType = "Node"
+	POD  EntityType = "Pod"
+	APP  EntityType = "App"
+)
+type ResourceMetric interface {
+	GetMetric() MetricPropType
+	GetResourceType() ResourceType
+	SetValue(data *float64)
+	GetValue() *float64
+}
+
+// ====================== Entity Resource Metric ===========================
+
+type EntityResourceMetrics struct {
+	entityId                string
+	ResourceMetricMap       map[string]ResourceMetric
+	boughtResourceMetricMap map[string]ResourceMetric
+}
+
+func NewEntityResourceMetrics(entityId string) *EntityResourceMetrics {
+	return &EntityResourceMetrics{
+		entityId: entityId,
+		ResourceMetricMap: make(map[string]ResourceMetric),
+	}
+}
+func (store *EntityResourceMetrics) SetResourceMetric(resource ResourceMetric) {
```

Comment:
if this is a store, we may want to rename the struct to sth like EntityResourceMetricsStore, to make it consistent.